appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java

```diff
@@ -153,112 +153,82 @@ public void accept(ApiVisitor visitor) {
+        processAnnotations(Extension.class, visitor::visitExtension, Extensions.class);
         processAnnotations(Operation.class, visitor::visitOperation);
-        processAnnotations(Callback.class, visitor::visitCallback, Callbacks.class, visitor::visitCallbacks);
-        processAnnotations(APIResponse.class, visitor::visitAPIResponse, APIResponses.class,
-                visitor::visitAPIResponses, true);
-        processAnnotations(Parameters.class, visitor::visitParameters, true);
-        processAnnotations(Parameter.class, visitor::visitParameter, true);
-        processAnnotations(ExternalDocumentation.class, visitor::visitExternalDocumentation, true);
-        processAnnotations(Tag.class, visitor::visitTag, Tags.class, visitor::visitTags, true);
-        processAnnotations(SecurityScheme.class, visitor::visitSecurityScheme, SecuritySchemes.class,
-                visitor::visitSecuritySchemes, true);
-        processAnnotations(SecurityRequirement.class, visitor::visitSecurityRequirement, SecurityRequirements.class,
-                visitor::visitSecurityRequirements, true);
+        processAnnotations(Callback.class, visitor::visitCallback, Callbacks.class);
+        processAnnotations(Callbacks.class, visitor::visitCallbacks, Callback.class);
+        processAnnotations(APIResponse.class, visitor::visitAPIResponse, APIResponses.class);
+        processAnnotations(APIResponses.class, visitor::visitAPIResponses, APIResponse.class);
+        processAnnotations(Parameters.class, visitor::visitParameters, Parameter.class);
+        processAnnotations(Parameter.class, visitor::visitParameter, Parameters.class);
+        processAnnotations(ExternalDocumentation.class, visitor::visitExternalDocumentation);
+        processAnnotations(Tag.class, visitor::visitTag, Tags.class);
+        processAnnotations(Tags.class, visitor::visitTags, Tag.class);
+        processAnnotations(SecurityScheme.class, visitor::visitSecurityScheme, SecuritySchemes.class);
+        processAnnotations(SecuritySchemes.class, visitor::visitSecuritySchemes, SecurityScheme.class);
+        processAnnotations(SecurityRequirement.class, visitor::visitSecurityRequirement, SecurityRequirements.class);
+        processAnnotations(SecurityRequirements.class, visitor::visitSecurityRequirements, SecurityRequirement.class);
 
         // JAX-RS response types
         processAnnotations(Produces.class, visitor::visitProduces);
         processAnnotations(Consumes.class, visitor::visitConsumes);
+
+        // OpenAPI response types
         processAnnotations(RequestBody.class, visitor::visitRequestBody);
     }
 
-    private <A extends Annotation, A2 extends Annotation, E extends AnnotatedElement> void processAnnotations(
-            Class<A> annotationClass, VisitorFunction<A, E> annotationFunction, Class<A2> altClass,
-            VisitorFunction<A2, E> altFunction, boolean allowInterfaces) {
```

Comment:
FYI: Interface exclusion are now handled elsewhere since interfaces are not disallowed generally since one can put annotations on the interface to make them take effect. With this this flag was no longer needed.